MarkBind/markbind
# update workflow and project structure of devGuide

**What is the purpose of this pull request?**




**Overview of changes:**
Fixes #1707
- Add a tooltip that includes a link to Babel's readme on the pros and cons of a monorepo
- Update the project structure section to include some more details about the external packages, pointing out those that have been patched
- Update some points for consideration when adding dependencies, under workflow

Potentially fixes #1012
 - The issue asks for a list of markdown-it plugins & rules to be included in the DG. I have added some descriptions about existing markdown-it plugins and also refer to the `package.json` for a full list. I feel that this might just be enough without providing too many details as to which exact rules/plugins are used, which can be subjected to (somewhat frequent) changes across versions. However, if there is still merit, I shall remove this part.

**Anything you'd like to highlight / discuss:**
- The above point on whether this PR is sufficient to close the related issue.
- Any proofreading, thoughts on the added content is appreciated.

**Testing instructions:**
See the two pages in the dev guide that have been updated
- [workflow](https://deploy-preview-1786--markbind-master.netlify.app/devguide/workflow#points-for-consideration)
- [project structure](https://deploy-preview-1786--markbind-master.netlify.app/devguide/design/projectstructure)


**Proposed commit message: (wrap lines at 72 characters)**
Update workflow and project structure of devGuide



---

**Checklist:** :ballot_box_with_check:
